contracts/staking/contracts/src/stake/MixinDelegatedStake.sol

```diff
@@ -0,0 +1,169 @@
+    IStakingEvents,
+    MixinDeploymentConstants,
+    MixinConstants,
+    MixinStorage,
+    MixinOwnable,
+    MixinScheduler,
+    MixinStakingPoolRewardVault,
+    MixinZrxVault,
+    MixinStakingPool,
+    MixinTimelockedStake,
+    MixinStakeBalances,
+    MixinStake,
+    MixinStakingPoolRewards
+{
+
+    /// @dev This mixin contains logic for managing delegated stake.
+    /// **** Read MixinStake before continuing ****
+    /// Stake can be delegated to staking pools in order to trustlessly
+    /// leverage the weight of several stakers. The meaning of this
+    /// leverage depends on the context in which stake the is being utilized.
+    /// For example, the amount of fee-based rewards a market maker receives
+    /// is correlated to how much stake has been delegated to their pool (see MixinExchangeFees).
+
+    using LibSafeMath for uint256;
+
+    /// @dev Deposit Zrx and mint stake in the "Activated & Delegated" state.
+    /// Note that the sender must be payable, as they may receive rewards in ETH from their staking pool.
+    /// @param poolId Unique Id of staking pool to delegate stake to.
+    /// @param amount of Zrx to deposit / Stake to mint.
+    function depositZrxAndDelegateToStakingPool(bytes32 poolId, uint256 amount)
+        external
+    {
+        address payable owner = msg.sender;
+        _mintStake(owner, amount);
+        activateStake(amount);
+        _delegateStake(owner, poolId, amount);
+    }
+
+    /// @dev Activates stake that is presently in the Deactivated & Withdrawable state.
+    /// Note that the sender must be payable, as they may receive rewards in ETH from their staking pool.
+    /// The newly activated stake is then delegated to a staking pool.
+    /// @param poolId Unique Id of staking pool to delegate stake to.
+    /// @param amount of Stake to activate & delegate.
+    function activateAndDelegateStake(
+        bytes32 poolId,
+        uint256 amount
+    )
+        public
+    {
+        activateStake(amount);
+        address payable owner = msg.sender;
+        _delegateStake(owner, poolId, amount);
+    }
+
+    /// @dev Deactivate & Timelock stake that is currently in the Activated & Delegated state.
+    /// Note that the sender must be payable, as they may receive rewards in ETH from their staking pool.
+    /// @param poolId Unique Id of staking pool that the Stake is currently delegated to.
+    /// @param amount of Stake to deactivate and timelock.
+    function deactivateAndTimelockDelegatedStake(bytes32 poolId, uint256 amount)
+        public
+    {
+        deactivateAndTimelockStake(amount);
+        address payable owner = msg.sender;
+        _undelegateStake(owner, poolId, amount);
+    }
+
+    /// @dev Delegates stake from `owner` to the staking pool with id `poolId`
+    /// @param owner of Stake
+    /// @param poolId Unique Id of staking pool to delegate stake to.
+    /// @param amount of Stake to delegate.
+    function _delegateStake(
```

Comment:
This delegation begins during the current epoch, right? If so, the stake needs to be somehow time-weighted to prevent users from delegating near the end of an epoch with reduced risk.

Reply:
Note: we discussed offline and have a task for this.